lib/filestorage.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const common = require('metarhia-common');
+const { crc32 } = require('crc');
+const { zip, gzip } = require('compressing');
+
+const FS_EXT = 'f';
+
+class FileStorage {
+
+  // Create new FileStorage
+  //   dir - <string>, data storage directory
+  constructor(dir) {
+    this.dir = path.resolve(dir);
+    this.compressionList = [];
+    if (!fs.existsSync(dir)) fs.mkdirSync(dir);
```

Comment:
I think we should expose some sort of 
```
create(dir, callback) {
  ...
}
```
function (not on FileStorage, just export from this file) that will check and create the directory if needed. As IMO using sync functions is not a good idea here.